### PR TITLE
Cloud Docker 1.2.0 Release

### PR DESCRIPTION
**Pull request by meker12**
## Purpose of this pull request

Adds release notes and documentation updates for Magento Cloud Docker 1.2.0 release.

## Affected DevDocs pages

-  https://devdocs.magento.com/cloud/docker/docker-config.html
-  https://devdocs.magento.com/cloud/docker/docker-containers-service.html
-  https://devdocs.magento.com/cloud/docker/docker-containers.html
-  https://devdocs.magento.com/cloud/docker/docker-development-debug.html
-  https://devdocs.magento.com/cloud/docker/docker-mode-developer.html
-  https://devdocs.magento.com/cloud/docker/docker-mode-production.html
-  https://devdocs.magento.com/cloud/docker/docker-quick-reference.html
-  https://devdocs.magento.com/cloud/docker/docker-test-app-mftf.html 
-  https://devdocs.magento.com/cloud/release-notes/mcd-release-notes.html

whatsnew
[Magento Cloud Docker 1.2.0 release notes](https://devdocs.magento.com/cloud/release-notes/mcd-release-notes.html) and updates:

-  Updated the [Set up email](https://devdocs.magento.com/cloud/docker/docker-config.html#set-up-email) topic to add instructions for configuring HTTP and SMTP ports and disabling MailHog in a Magento Cloud Docker environment.

- Added instructions for configuring the Docker host name for the Xdebug container when running Magento Cloud Docker on Linux.

-  Added instructions for updating the [Elasticsearch plugin configuration](/cloud/docker/docker-containers-service.html#elasticsearch-plugins) in a Magento Cloud Docker environment.

- Added instructions for [building a custom `docker-compose.yaml` file](https://devdocs.magento.com/cloud/docker/docker-quick-reference.html#build-a-custom-docker-compose-configuration).

### CODE REVIEW

File: src/cloud/docker/docker-quick-reference.md [hunk 1]
```diff
@@ -10,6 +10,8 @@ functional_areas:
 
 ## Docker Compose
 
+Docker Compose is a tool for defining and running multi-container Docker applications. The following table lists the Docker Compose commands for building, deploying, and operating {{site.data.var.mcd-prod}}. You can also use [Magento Cloud Docker CLI](#magento-cloud-docker-cli) commands to complete Docker Compose tasks.
```
Comment (meker12) [inline src/cloud/docker/docker-quick-reference.md:13]:
```suggestion
Docker Compose is a tool for defining and running multi-container Docker applications. The following table lists the `docker-compose` commands for building, deploying, and operating {{site.data.var.mcd-prod}}. You can also use [Magento Cloud Docker CLI](#magento-cloud-docker-cli) commands to complete Docker Compose tasks.
```

File: src/cloud/release-notes/mcd-release-notes.md [hunk 2]
```diff
@@ -16,6 +16,53 @@ The release notes include:
...
+
+      -  **Elasticsearch plugin support**–Added support for the following Elasticsearch plugins: `analysis-icu`, `analysis-phonetic`, `analysis-stempel`, and `analysis-nori`.  The `analysis-icu` and `analysis-phonetic` plugins are installed by default. You can add or remove the `analysis-stempel` and `analysis-nori` plugins as needed.<!--MCLOUD-2789-->
+
+   -  {:.new} **CLI container**
+
+      -  **Run commands inside Docker PHP containers**–Now you can use the Magento Cloud Docker CLI to run commands inside PHP containers in your Docker environment without having to install PHP on the host. For example, the following command builds the configuration:  `./bin/magento-docker php 7.3 vendor/bin/ece-docker build:compose`. See [Magento Cloud Docker CLI]({{ site.baseurl }}/cloud/docker/docker-quick-refere.html#magento-cloud-dockencer-cli). *[Fix submitted by G Arvind from Zilker Technology](https://github.com/magento/magento-cloud-docker/pull/209).*<!--MCLOUD-5982-->
```
Comment (meker12) [inline src/cloud/release-notes/mcd-release-notes.md:38]:
```suggestion
      -  **Run commands inside Docker PHP containers**–Now you can use the Magento Cloud Docker CLI to run commands inside PHP containers in your Docker environment without having to install PHP on the host. For example, the following command builds the configuration:  `./bin/magento-docker php 7.3 vendor/bin/ece-docker build:compose`. See [Magento Cloud Docker CLI]({{ site.baseurl }}/cloud/docker/docker-quick-refere.html#magento-cloud-docker-cli). *[Fix submitted by G Arvind from Zilker Technology](https://github.com/magento/magento-cloud-docker/pull/209).*<!--MCLOUD-5982-->
```

File: src/cloud/release-notes/mcd-release-notes.md [hunk 3]
```diff
@@ -16,6 +16,53 @@ The release notes include:
...
+
+      -  **Elasticsearch plugin support**–Added support for the following Elasticsearch plugins: `analysis-icu`, `analysis-phonetic`, `analysis-stempel`, and `analysis-nori`.  The `analysis-icu` and `analysis-phonetic` plugins are installed by default. You can add or remove the `analysis-stempel` and `analysis-nori` plugins as needed.<!--MCLOUD-2789-->
+
+   -  {:.new} **CLI container**
+
+      -  **Run commands inside Docker PHP containers**–Now you can use the Magento Cloud Docker CLI to run commands inside PHP containers in your Docker environment without having to install PHP on the host. For example, the following command builds the configuration:  `./bin/magento-docker php 7.3 vendor/bin/ece-docker build:compose`. See [Magento Cloud Docker CLI]({{ site.baseurl }}/cloud/docker/docker-quick-refere.html#magento-cloud-docker-cli). *[Fix submitted by G Arvind from Zilker Technology](https://github.com/magento/magento-cloud-docker/pull/209).*<!--MCLOUD-5982-->
```
Comment (meker12) [inline src/cloud/release-notes/mcd-release-notes.md:38]:
```suggestion
      -  **Run commands inside Docker PHP containers**–Now you can use the Magento Cloud Docker CLI to run commands inside PHP containers in your Docker environment without having to install PHP on the host. For example, the following command builds the configuration:  `./bin/magento-docker php 7.3 vendor/bin/ece-docker build:compose`. See [Magento Cloud Docker CLI]({{ site.baseurl }}/cloud/docker/docker-quick-reference.html#magento-cloud-docker-cli). *[Fix submitted by G Arvind from Zilker Technology](https://github.com/magento/magento-cloud-docker/pull/209).*<!--MCLOUD-5982-->
```

**Comment by andriyShevtsov** [comment]
QA approved

File: src/cloud/docker/docker-quick-reference.md [hunk 4]
```diff
@@ -46,14 +48,266 @@ docker-compose -f docker-compose.yml -f docker-compose-custom.yml [-f more-custo
...
+Instead of building the `docker-compose.yaml` file using the {{site.data.var.ece }} project configuration. You can use the `build:custom:compose` command to build a custom `docker-compose.yaml` file with the configuration you supply. You provide the configuration as a JSON array as shown in [Example 1](#example-1).
+
+For {{site.data.var.mcd-prod}} 1.2 and later, you have an additional option to specify custom images and image versions using the `ece-docker build:custom:compose` command as shown in [Example 2](#example-2).
+
+#### Example 1
+
+> Generate a custom `docker-compose.yaml` file
+
+```bash
+./vendor/bin/ece-docker build:custom:compose '{"name":"magento","system":{"mode":"production","host":"magento2.test","port":"8080","db":{"increment_increment":3,"increment_offset":2},"mailhog":{"smtp_port":"1026","http_port":"8026"}},"services":{"php":{"version":"7.4","enabled":true,"extensions":{"enabled":["xsl"]}},"mysql":{"version":"10.2","image":"mariadb","enabled":true}, "mailhog": {"enabled":true}}}'
```
Comment (meker12) [inline src/cloud/docker/docker-quick-reference.md:125]:
```suggestion
./vendor/bin/ece-docker build:custom:compose '{"name":"magento","system":{"mode":"production","host":"magento2.test","port":"8080","db":{"increment_increment":3,"increment_offset":2},"mailhog":{"smtp_port":"1026","http_port":"8026"}},"services":{"php":{"version":"7.3","enabled":true,"extensions":{"enabled":["xsl"]}},"mysql":{"version":"10.2","image":"mariadb","enabled":true}, "mailhog": {"enabled":true}}}'
```

File: src/cloud/docker/docker-quick-reference.md [hunk 5]
```diff
@@ -46,14 +48,266 @@ docker-compose -f docker-compose.yml -f docker-compose-custom.yml [-f more-custo
...
+  magento-magento-db: {  }
+networks:
+  magento:
+    driver: bridge
+  magento-build:
+    driver: bridge
+
+```
+
+For {{site.data.var.mcd-prod}} 1.2 and later, you have an additional option to specify custom images and image versions using the `ece-docker build:custom:compose` command.
```
Comment (meker12) [inline src/cloud/docker/docker-quick-reference.md:276]:
```suggestion
```

File: src/cloud/release-notes/mcd-release-notes.md [hunk 6]
```diff
@@ -16,6 +16,53 @@ The release notes include:
...
+
+      -  **Run commands inside Docker PHP containers**–Now you can use the Magento Cloud Docker CLI to run commands inside PHP containers in your Docker environment without having to install PHP on the host. For example, the following command builds the configuration:  `./bin/magento-docker php 7.3 vendor/bin/ece-docker build:compose`. See [Magento Cloud Docker CLI]({{ site.baseurl }}/cloud/docker/docker-quick-reference.html#magento-cloud-docker-cli). *[Fix submitted by G Arvind from Zilker Technology](https://github.com/magento/magento-cloud-docker/pull/209).*<!--MCLOUD-5982-->
+
+      -  Added the OpenSSH-client to PHP CLI containers so you can use ssh-agent forwarding for Composer if the `composer.json` file contains private git repositories that require an ssh client for various Composer commands.<!--MCLOUD-6008-->
+
+   -  {:.new}**Test container**–Added a test container for Magento application testing and added the `--with-test` option to the Docker `build:compose` command to create the container only when testing Magento in the Docker environment. See [Magento application testing](https://devdocs.magento.com/cloud/docker/docker-test-app-mftf.html).<!--MCLOUD-6394-->
```
Comment (hguthrie) [inline src/cloud/release-notes/mcd-release-notes.md:42]:
```suggestion
   -  {:.new}**Test container**–Added a test container for Magento application testing, and added the `--with-test` option to the Docker `build:compose` command to create the container only when testing Magento in the Docker environment. See [Magento application testing](https://devdocs.magento.com/cloud/docker/docker-test-app-mftf.html).<!--MCLOUD-6394-->
```

File: src/cloud/release-notes/mcd-release-notes.md [hunk 7]
```diff
@@ -16,6 +16,53 @@ The release notes include:
...
+
+   -  For {{site.data.var.mcd-prod}} 1.2.0 and later, Magento now provides Docker images for each patch version, and the Docker configuration generator creates the Docker configuration with a specified patch version instead of using the latest. Previously, the Docker configuration generator built the configuration using the latest patch version which could break {{ site.data.var.mcd-prod}} environments built using an earlier version.<!--MCLOUD-7093-->
+
+   -  **Specify custom images and versions in custom Magento Cloud Docker configuration**service configuration  custom environment Updated the `build:custom:compose` command with options to specify custom images and versions when generating a custum Docker compose configuration file, `docker-compose.yaml`. See [Build a custom Docker Compose configuration]({{ site.baseurl }}/cloud/docker/docker-quick-reference.html#build-a-custom-docker-compose-configuration). <!--MCLOUD-7089-->
```
Comment (hguthrie) [inline src/cloud/release-notes/mcd-release-notes.md:56]: Something is not finished in this entry. Spacing problems, misspelling of custom, and other issues. May need review, but I'll give it a go.

```suggestion
   -  **Specify custom images and versions in custom Magento Cloud Docker configuration**—Updated the `build:custom:compose` command with options to specify custom images and versions when generating a custom Docker compose configuration file (`docker-compose.yaml`). See [Build a custom Docker Compose configuration]({{ site.baseurl }}/cloud/docker/docker-quick-reference.html#build-a-custom-docker-compose-configuration). <!--MCLOUD-7089-->
```

**Review by hguthrie — CHANGES_REQUESTED** [review]
A few problems I noted. Also not part of your changes, but I noticed Line 180 of containers topic (selenium) looks like it has a broken link. I think it needs a closing paren.

File: src/cloud/docker/docker-quick-reference.md [hunk 8]
```diff
@@ -46,14 +48,264 @@ docker-compose -f docker-compose.yml -f docker-compose-custom.yml [-f more-custo
...
+      --no-ansi                                        Disable ANSI output
+  -n, --no-interaction                                 Do not ask any interactive question
+  -v|vv|vvv, --verbose                                 Increase the verbosity of messages: 1 for normal output, 2 for more verbose output and 3 for debug
+  ```
 
 {:.bs-callout-info}
-See [Service versions] for a list of service configuration options you can add to the `ece-docker build:compose` command to customize the Cloud Docker environment configuration.
+See [Service versions] for additional information about the service configuration options for the `ece-docker build:compose` command.
+
+### Build a custom Docker Compose configuration
```
[the remark this answers is not in the document]
Comment (meker12) [reply src/cloud/docker/docker-quick-reference.md:114]: Leaving for now

File: src/cloud/docker/docker-containers-service.md [hunk 9]
```diff
@@ -168,6 +182,14 @@ docker-compose run --rm redis redis-cli -h redis
 
 The Selenium container, based on the [selenium/standalone-chrome/](https://hub.docker.com/r/selenium/standalone-chrome/h), enables the [Magento Functional Testing Framework (MFTF)](https://devdocs.magento.com/mftf/docs/introduction.html) for Magento application testing in the Cloud Docker environment. See [Magento application testing]({{site.baseurl}}/cloud/docker/docker-test-app-mftf.html).
 
+## Test container
+
+**Container name**: test<br/>
+**Docker base image**: [magento/magento-cloud-docker-php][php-cloud], which is based on the [php](https://hub.docker.com/_/php) Docker image<br/>
```
[the remark this answers is not in the document]
Comment (meker12) [reply src/cloud/docker/docker-containers-service.md]: Handled inconsistently. Moved "based on ... Docker image" to the header info for all containers.

**Review by hguthrie — APPROVED** [review]
Ship it!

**Pull request merged**